document/field/field.go

```diff
@@ -6,6 +6,24 @@ type Field struct {
 	Value ValueUnion
 }
 
+// Reset resets a field.
+func (f *Field) Reset() {
+	f.Path = nil
+	f.Value = ValueUnion{}
+}
+
+// ReturnArrayToPool returns a document array to pool.
+func ReturnArrayToPool(fields []Field, p *BucketizedFieldArrayPool) {
+	if p == nil {
+		return
+	}
+	for i := 0; i < len(fields); i++ {
+		fields[i].Reset()
+	}
+	fields = fields[:0]
+	p.Put(fields, cap(fields))
```

Comment:
nit: is it better to change the Put method to just take in fields? so the user don't need to decide whether to use `cap` or `len` for the size, the `Put` method can call `cap` itself.

Reply:
Yeah so unfortunately this is dictated by the template as the array type is a `GenericValue` in the template, and does not have a `cap` method.